src/pint/pintk/plk.py

```diff
@@ -524,6 +524,39 @@ def setChoice(self, xid="mjd", yid="pre-fit"):
             if choice.lower() == yid:
                 self.ybuttons[ii].select()
 
+            if choice == "elongation":
+                if not any(
+                    x.startswith("Astrometry")
+                    for x in self.master.psr.prefit_model.components
+                ):
+                    self.xbuttons[ii].configure(state="disabled")
+                    if not self.master.psr.fitted:
+                        self.ybuttons[ii].configure(state="disabled")
+                if self.master.psr.fitted and not any(
+                    x.startswith("Astrometry")
+                    for x in self.master.psr.pstfit_model.components
+                ):
+                    self.ybuttons[ii].configure(state="disabled")
+            if choice == "orbital phase":
+                if not any(
+                    x.startswith("Binary")
+                    for x in self.master.psr.prefit_model.components
+                ):
+                    self.xbuttons[ii].configure(state="disabled")
+                    if not self.master.psr.fitted:
+                        self.ybuttons[ii].configure(state="disabled")
+                if self.master.psr.fitted and not any(
+                    x.startswith("Binary")
+                    for x in self.master.psr.prefit_model.components
+                ):
+                    self.ybuttons[ii].configure(state="disabled")
+            if choice == "frequency" and (
+                (len(np.unique(self.master.psr.all_toas["freq"])) <= 1)
+                or np.any(np.isinf(self.master.psr.all_toas["freq"]))
```

Comment:
Shoudn't this be 
`np.all(np.isfinite(self.master.psr.all_toas["freq"]))`
?

Reply:
Isn't that the same thing?  `any(isinf)` vs `all(finite)`?

Reply:
No. any(isinf) == not all(finite)
(de Morgan's identity)

Reply:
but here we are excluding data with any infinite frequencies.

Reply:
OK.. My bad. I thought the code was conditionally enabling the buttons instead of disabling them.